lib/zilliqa/account/wallet.rb

```diff
@@ -135,10 +146,24 @@ def sign_with(tx, address)
         end
 
         tx.sender_pub_key = account.public_key
-        sig = account.sign_transaction(tx)
+
+        begin
+          sig ||= account.sign_transaction(tx)
+          raise InvalidSignatureSizeError.new(sig.to_s) unless valid_signature?(sig)
```

Comment:
Wouldn't `if` statement here be more effective? `raise` & `rescue` are very resource consuming.

Reply:
Put it into Schnorr class